### DISCUSSION TITLE
Neighbors

### DISCUSSION THREAD

**Discussion by kmfnjtze8888** [openthread/openthread#8636] [Q&A] [answered]
Hello,

Is there a way to set a minimum RSSI threshold for neighbors so a device will only pick neighbors with RSSI values at or above this value?

I currently have a border router that for some reason picks neighbors with poor RSSI even though there are full-thread devices that are closer and have better RSSI. This makes all the units unable to send data to the border router.

**neighbor table**
![image](https://user-images.githubusercontent.com/72487638/211366378-14550980-d482-49f1-8c43-5b616ede49ab.png)

**scan**
![image](https://user-images.githubusercontent.com/72487638/211366728-96472479-fdff-4e34-86ad-cc3455f3ac3d.png)

Thanks.

**Comment by jwhui** [comment, marked as answer]
Routers do not attempt to establish links with neighboring routers when the link margin is below threshold, as configured by: https://github.com/openthread/openthread/blob/fd9f3237eb525fff6cfe5a6af54fb0bdc0c86f72/src/core/config/mle.h#L203

Link margin is determined by radio's receive sensitivity and receive signal strength for a given received packet:
https://github.com/openthread/openthread/blob/fd9f3237eb525fff6cfe5a6af54fb0bdc0c86f72/include/openthread/platform/radio.h#L442

https://github.com/openthread/openthread/blob/fd9f3237eb525fff6cfe5a6af54fb0bdc0c86f72/include/openthread/platform/radio.h#L317

I would validate that the radio is properly reporting the radio receive sensitivity and signal strength.

**Reply by kmfnjtze8888** [reply]
> I would validate that the radio is properly reporting the radio receive sensitivity and signal strength.

Sorry, I am not very experienced at this. What are the common ways I can do to validate these?

**Reply by jwhui** [reply]
Doing this properly typically requires some instrumentation, such as a power meter.
